src/server/rpc/expr.rs

```diff
@@ -445,6 +481,121 @@ fn build_binary_expr(op: Operator, inputs: Vec<Expr>) -> Result<Expr> {
     }
 }
 
+pub fn make_read_group_aggregate(
+    aggregate: Option<RPCAggregate>,
+    _group: RPCGroup,
+    group_keys: Vec<String>,
+) -> Result<GroupByAndAggregate> {
+    let gby_agg = GroupByAndAggregate::Columns {
+        agg: convert_aggregate(aggregate)?,
+        group_columns: group_keys,
+    };
+    Ok(gby_agg)
+}
+
+/// Builds GroupByAndAggregate::Windows
+pub fn make_read_window_aggregate(
+    aggregates: Vec<RPCAggregate>,
+    window_every: i64,
+    offset: i64,
+    window: Option<RPCWindow>,
+) -> Result<GroupByAndAggregate> {
+    // only support single aggregate for now
+    if aggregates.len() != 1 {
+        return AggregateNotSingleton { aggregates }.fail();
+    }
+    let agg = convert_aggregate(Some(aggregates.into_iter().next().unwrap()))?;
```

Comment:
Do we want to explicitly panic if the vec is empty? If not, how about:

```suggestion
    let agg = convert_aggregate(aggregates.into_iter().next())?;
```

Reply:
I think check above for `aggregates.len() == 1` covers the empty case, but in general I don't like this pattern to de-"destruture" a Vec, but sadly I don't know any better way

However the change to avoid unwrap is nice -- I'll do that in a follow on PR